src/Icicle/Sea/FromAvalanche/Psv.hs

```diff
@@ -828,163 +851,132 @@ seaOfWriteOutput ps oname@(OutputName name) otype0 ts0 ixStart
     hasDateTime (StructT m) = any hasDateTime $ Map.elems $ getStructType m
     hasDateTime _           = False
 
-seaOfOutput
-  :: Doc -> OutputName -> ValType -> [ValType] -> Int
-  -> Doc                                        -- buffer to print to
-  -> Either SeaError (Doc, [(BufSize, BufLen)]) -- name, alloc size and len
-
-seaOfOutput ps oname@(OutputName name) otype0 ts0 ixStart dstbuf
+seaOfOutput :: Doc -> OutputName -> ValType -> [ValType] -> Int -> Either SeaError Doc
+seaOfOutput ps oname@(OutputName name) otype0 ts0 ixStart
```

Comment:
In the end this should really return the `[ValType]` that was not consumed so it can be called recursively like [unmeltValue'](https://github.com/ambiata/icicle/blob/e68df32f1cc536aac783c3bbbcf2e031c10f98ae/src/Icicle/Avalanche/Prim/Eval.hs#L287)